toolchain/check/import_cpp.cpp

```diff
@@ -44,6 +44,71 @@ static auto GenerateCppIncludesHeaderCode(
   return code;
 }
 
+namespace {
+
+class CarbonClangDiagnosticConsumer : public clang::DiagnosticConsumer {
+ public:
+  CarbonClangDiagnosticConsumer(Context& context, SemIRLoc loc)
+      : context_(context), loc_(loc) {}
+
+  void HandleDiagnostic(clang::DiagnosticsEngine::Level diag_level,
+                        const clang::Diagnostic& info) override {
+    DiagnosticConsumer::HandleDiagnostic(diag_level, info);
+
+    llvm::SmallString<100> message;
+    info.FormatDiagnostic(message);
+
+    std::string diagnostics_str;
+    llvm::raw_string_ostream diagnostics_stream(diagnostics_str);
+    clang::TextDiagnostic text_diagnostic(
+        diagnostics_stream,
+        // TODO: Consider allowing setting `LangOptions` or use
+        // `ASTContext::getLangOptions()`.
+        clang::LangOptions(),
+        // TODO: Consider allowing setting `DiagnosticOptions` or use
+        // `ASTUnit::getDiagnostics().::getLangOptions().getDiagnosticOptions()`.
+        new clang::DiagnosticOptions());
+    text_diagnostic.emitDiagnostic(
+        clang::FullSourceLoc(info.getLocation(), info.getSourceManager()),
+        diag_level, message, info.getRanges(), info.getFixItHints());
+
+    switch (diag_level) {
+      case clang::DiagnosticsEngine::Ignored:
+      case clang::DiagnosticsEngine::Note:
+      case clang::DiagnosticsEngine::Remark: {
+        context_.TODO(
+            loc_, llvm::formatv(
+                      "Unsupported: C++ diagnostic level for diagnostic {0}",
```

Comment:
```suggestion
                      "Unsupported: C++ diagnostic level for diagnostic\n{0}",
```
The diagnostic message may be pretty long, so perhaps give it a linebreak as a separator?

Reply:
Done.